src/Parsers/ASTInsertQuery.cpp

```diff
@@ -86,33 +86,41 @@ void ASTInsertQuery::formatImpl(const FormatSettings & settings, FormatState & s
             settings.ostr << (settings.hilite ? hilite_keyword : "") << " COMPRESSION " << (settings.hilite ? hilite_none : "") << compression->as<ASTLiteral &>().value.safeGet<std::string>();
     }
 
+    if (settings_ast)
+    {
+        settings.ostr << (settings.hilite ? hilite_keyword : "") << settings.nl_or_ws << "SETTINGS " << (settings.hilite ? hilite_none : "");
+        settings_ast->formatImpl(settings, state, frame);
+        settings.ostr << settings.nl_or_ws;
```

Comment:
It seems if we replace this line with

```
space = settings.nl_or_ws
```

then the conditions below 
```
        if (!settings_ast)
            settings.ostr << ' ';
```

will turn into just `settings.ostr << space`

Reply:
This will break compatibility for INSERT w/o SETTINGS formatted in oneline, but I got the idea, will simplify the code.